src/gamebryo/gamebryosavegame.cpp

```diff
@@ -151,29 +163,29 @@ void GamebryoSaveGame::FileWrapper::read(void *buff, std::size_t length)
   }
 }
 
-void GamebryoSaveGame::FileWrapper::readImage(int scale, bool alpha)
+QImage GamebryoSaveGame::FileWrapper::readImage(int scale, bool alpha)
 {
   unsigned long width;
   read(width);
   unsigned long height;
   read(height);
-  readImage(width, height, scale, alpha);
+  return readImage(width, height, scale, alpha);
 }
 
-void GamebryoSaveGame::FileWrapper::readImage(unsigned long width, unsigned long height, int scale, bool alpha)
+QImage GamebryoSaveGame::FileWrapper::readImage(unsigned long width, unsigned long height, int scale, bool alpha)
 {
   int bpp = alpha ? 4 : 3;
   QScopedArrayPointer<unsigned char> buffer(new unsigned char[width * height * bpp]);
   read(buffer.data(), width * height * bpp);
   QImage image(buffer.data(), width, height, alpha ? QImage::Format_RGBA8888_Premultiplied
     : QImage::Format_RGB888);
   if (scale != 0) {
-    m_Game->m_Screenshot = image.copy().scaledToWidth(scale);
+    return image.copy().scaledToWidth(scale);
   } else {
     // why do I have to copy here? without the copy, the buffer seems to get
```

Comment:
Can you clean up the comment? `QImage` doesn't copy the buffer you give it and the buffer dies at the end of the function.